r/vignettes/arrow.Rmd

```diff
@@ -86,7 +88,73 @@ to other applications and services that use Arrow. One example is Spark: the
 move data to and from Spark, yielding [significant performance
 gains](http://arrow.apache.org/blog/2019/01/25/r-spark-improvements/).
 
-# Class structure and package conventions
+# Internals
+
+## Mapping of R <--> Arrow types
+
+Arrow has a rich data type system that includes direct parallels with R's data types and much more.
+
+In the tables, entries with a `-` are not currently implemented.
+
+### R to Arrow
+
+| R type                   | Arrow type |
+|--------------------------|------------|
+| logical                  | boolean    |
+| integer                  | int32      |
+| double ("numeric")       | float64    |
+| character                | utf8       |
+| factor                   | dictionary |
+| raw                      | uint8      |
+| Date                     | date32     |
+| POSIXct                  | timestamp  |
+| POSIXlt                  | -          |
+| data.frame               | struct     |
+| list^+^                  | list       |
+| bit64::integer64         | int64      |
+| difftime                 | time32     |
+| vctrs::vctrs_unspecified | null       |
+
+^+^: Only lists where all elements are the same type are able to be translated to Arrow list type (which is a "list of" some type).
+
+### Arrow to R
+
+| Arrow type        | R type                   |
+|-------------------|--------------------------|
+| boolean           | logical                  |
+| int8              | integer                  |
+| int16             | integer                  |
+| int32             | integer                  |
+| int64             | bit64::integer64         |
+| uint8             | integer                  |
+| uint16            | integer                  |
+| uint32            | double                   |
```

Comment:
I'm curious about the double, is it a copy-paste error, or it really casts to double.

Reply:
It's "real" ;) https://github.com/apache/arrow/blob/master/r/src/array_to_vector.cpp#L717-L720

Reply:
@romainfrancois, why not using the same as int64_t?

Reply:
Perhaps we could. The `bit64::integer64` class we use for int64 only supports signed integers, so `uint32` could fit in that, though `uint64` could not (cf. https://jira.apache.org/jira/browse/ARROW-9031).

I would also like to [pull int64, uint32, and uint64 into an R integer](https://jira.apache.org/jira/browse/ARROW-9083) if they are in bounds (32-bit signed).

In any case, this PR is merely documenting the status quo, so if we want to change uint32 behavior, let's open a new JIRA.

Reply:
Expanded https://jira.apache.org/jira/browse/ARROW-9083 to cover uint{32,64}. I'm going to proceed with this PR as is and will happily revise the docs if we change the implementation. 

Thanks for the discussion--shining a light on things like this is a big reason I picked up the documentation task.